lib/derivatives.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require_relative "dates"
+
+module PostZephirProcessing
+  # A class that knows the expected locations of standard Zephir derivative files.
+  # `earliest_missing_date` is the main entrypoint when constructing an agenda of Zephir
+  # file dates to fetch for processing.
+  class Derivatives
+    DIR_DATA = {
+      zephir_full: {
+        name: :zephir_full,
+        location: :CATALOG_PREP,
```

Comment:
Alas, it's needed because the various tests run in their own temp directories, so I can't think of a clean way to keep that indirection out of the data structure.